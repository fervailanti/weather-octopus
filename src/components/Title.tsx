import React, { FC, ReactElement } from 'react'

interface TitleProps {
  text: {
    main: string
    side: string
  }
  color?: 'light' | 'dark'
  size?: 'small' | 'medium' | 'large'
}

const Title: FC<TitleProps> = ({ text, color = 'light', size = 'small' }): ReactElement => {
  const styles = { light: 'text-white', dark: 'text-gray-900' }
  const sizes = {
    main: {
      small: 'text-xl',
      medium: 'text-2xl',
      large: 'text-5xl'
    },
    side: {
      small: 'text-sm',
      medium: 'text-lg',
      large: 'text-2xl'
    }
  }
  return (
    <h2 className={styles[color]}>
      <span className={`font-semibold ${sizes.main[size]}`}>{text.main}</span>
      <span className={sizes.side[size]}>, {text.side}</span>
    </h2>
  )
}

export default Title
