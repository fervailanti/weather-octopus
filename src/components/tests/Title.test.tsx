import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Title from '../Title'

describe('Title Component', () => {
  it('Should render correctly', () => {
    const title = render(
      <Title main={{ text: 'New York', size: 'xl' }} side={{ text: 'USA', size: 'md' }} />
    )
    expect(title.container).toMatchSnapshot()
    expect(title.container).toHaveTextContent('New York, USA')
  })
})
