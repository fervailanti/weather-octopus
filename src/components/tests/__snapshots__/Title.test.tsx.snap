// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Should render correctly 1`] = `
<div>
  <h2
    class="text-white"
  >
    <span
      class="font-semibold text-xl"
    >
      New York
    </span>
    <span
      class="text-md"
    >
      , 
      USA
    </span>
  </h2>
</div>
`;
